### PR TITLE
User story 26

### LINKED ISSUE #57: User Story 26, Registered users can check out
```
As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out
An order is created in the system, which has a status of "pending"
I am taken to my orders page ("/profile/orders")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty
```

DO THESE STORIES FIRST:
- User Story 21, User views their cart show page with items in the cart

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #249

child of #274

depends on #247

### PR DESCRIPTION
**Pull request by VinceCarollo**
## What does this PR do?

- Adds feature test to create orders only if registered user on carts show page with items in cart
- Add create action to orders controller that works with cart poro
- Add route for clicking of checkout button to orders controller create action
- Adds create_order method to cart PORO to create order and iterate over cart contents to create order items
- Adds button to checkout cart to carts show page only if current user is registered

resolves #57

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -21,4 +21,12 @@ def ids_to_items
   def total_price
     ids_to_items.sum{|item, qty| item.price * qty}
   end
+
+  def create_order(buyer_id)
+    order = Order.create(user_id: buyer_id, status: 1)
+    ids_to_items.each do |item, qty|
+      oi = OrderItem.new(item: item, order: order, quantity: qty, price: item.price, fulfilled: false )
+      oi.save
```
Comment (zachlp2016) [inline app/models/cart.rb:29]: Awesome looks gouda.

**Review by zachlp2016 — APPROVED** [review]
It's Grate!
